src/FeedlyProvider.cpp

```diff
@@ -58,7 +58,17 @@ void FeedlyProvider::authenticateUser(){
 #ifdef __APPLE__
                 system(std::string("open \"https://feedly.com/v3/auth/dev\" &> /dev/null &").c_str());
 #else
-                system(std::string("xdg-open \"https://feedly.com/v3/auth/dev\" &> /dev/null &").c_str());
+                if(getenv("XDG_DATA_DIRS") != NULL){
```

Comment:
`XDG_DATA_DIRS` is an optional environment variable for XDG Base Directory Specification. It may not defined on other environments/distributions. Also, the environment variables of the XDG Base Directory Specification are no limited to desktop sessions, instead they can (and should) be defined on tty sessions.

Another solution would be to check `XDG_SESSION_TYPE` which is always set by Systemd. Possible values are "x11", "wayland", "mir" for desktop sessions and "tty", "unspecified" (for tty sessions)

Reply:
Thank you for letting me know the correct way to detect a desktop session.

I tried `XDG_SESSION_TYPE` but it didn't work because it's not set in Xfce Terminal on Ubuntu 18.04.1 LTS.  `XDG_DATA_DIRS` is the only available environment variable whose name starts with "XDG_".  That might be because I'm connecting to the Xfce session via xrdp and/or Ubuntu is hosted by Windows Subsystem for Linux 2.

Do you have any idea of other better alternatives?

Reply:
I think in this case, we can check if `DISPLAY` is set (X11 sessions) or if `WAYLAND_DISPLAY` is set (wayland sessions).

Reply:
Thank you.  `DISPLAY` and `WAYLAND_DISPLAY` should work (at least on my environment).

I found the real problem was that the output redirection was not working.  When the output is redirected, `w3m` should terminate immediately after outputting the content of the specified page.  I fixed this by removing `&`.  I confirmed `xdg-open` still existed immediately after launching the browser process.

I removed the desktop session check because a desktop session doesn't guarantee that the browser won't crash or output texts to the standard output.  Instead, I decided to output the authentication page URL as an alternative method in case of any browser launch issue.